src/VecSim/algorithms/hnsw/hnsw_serializer_declarations.h

```diff
@@ -21,7 +21,8 @@ void saveIndexFields(std::ofstream &output) const;
 void saveGraph(std::ofstream &output) const;
 
 void saveLevel(std::ofstream &output, LevelData &data) const;
-void restoreLevel(std::ifstream &input, LevelData &data);
+void restoreLevel(std::ifstream &input, LevelData &data, EncodingVersion version);
```

Comment:
The serializer should be initialized with a specific version. Consider using inheritance to expand capabilities